src/Admin/ReenableCssTransientCachingAjaxAction.php

```diff
@@ -56,53 +56,47 @@ public function register() {
 	 * @param string $hook_suffix Hook suffix to identify from what admin page the call is coming from.
 	 */
 	public function register_ajax_script( $hook_suffix ) {
-		if ( 'site-health.php' !== $hook_suffix ) {
+		$supported_hook_suffix = defined( 'HEALTH_CHECK_PLUGIN_VERSION' ) ? 'tools_page_health-check' : 'site-health.php';
+		if ( $supported_hook_suffix !== $hook_suffix ) {
 			return;
 		}
 
-		$script = <<< 'JS_SCRIPT'
-;( function () {
-	window.addEventListener( 'DOMContentLoaded', ( event ) => {
-		var selector = SELECTOR;
-		( document.querySelectorAll( selector ) || [] )
-			.forEach( ( element ) => {
-				element.addEventListener( 'click', function ( event ) {
-					event.preventDefault();
-					if ( element.classList.contains( 'disabled' ) ) {
-						return;
-					}
-					wp.ajax.post( ACTION, ARGUMENTS )
-						.done( function () {
-							element.classList.remove( 'ajax-failure' );
-							element.classList.add( 'ajax-success' )
-							element.classList.add( 'disabled' )
-						} )
-						.fail( function () {
-							element.classList.remove( 'ajax-success' );
-							element.classList.add( 'ajax-failure' )
-							element.classList.add( 'disabled' )
-						} );
+		$exports = [
+			'selector' => self::SELECTOR,
+			'action'   => self::AJAX_ACTION,
+			'postArgs' => [ 'nonce' => wp_create_nonce( self::AJAX_ACTION ) ],
+		];
+
+		ob_start();
+		?>
+		<script>
+			(( $, { selector, action, postArgs } ) => {
+				document.addEventListener( 'DOMContentLoaded', () => {
+					$( '.health-check-body' ).on( 'click', selector, ( event ) => {
```

Comment:
Oh cool workaround! And TIL you could pass a selector to a jQuery event handler to filter the element that triggered the event.

Reply:
My stoneage JS knowledge still holds some surprises I guess. 😊